crates/cxx-qt-build/src/lib.rs

```diff
@@ -186,15 +196,15 @@ fn generate_cxxqt_cpp_files(
     header_dir: impl AsRef<Path>,
 ) -> Vec<GeneratedCppFilePaths> {
     let manifest_dir = env::var("CARGO_MANIFEST_DIR").unwrap();
-    let cpp_directory = format!("{}/cxx-qt-gen/src", env::var("OUT_DIR").unwrap());
 
     let mut generated_file_paths: Vec<GeneratedCppFilePaths> = Vec::with_capacity(rs_source.len());
     for rs_path in rs_source {
+        let cpp_directory = format!("{}/cxx-qt-gen/src", env::var("OUT_DIR").unwrap());
         let path = format!("{}/{}", manifest_dir, rs_path.display());
         println!("cargo:rerun-if-changed={}", path);
 
         let generated_code = GeneratedCpp::new(&path);
-        generated_file_paths.push(generated_code.write_to_directories(&cpp_directory, &header_dir));
+        generated_file_paths.push(generated_code.write_to_directories(cpp_directory, &header_dir));
```

Comment:
eg maybe keep this one as a reference and move the `cpp_directory` out of the loop